src/pilot/assign_role.py

```diff
@@ -1160,11 +1168,46 @@ def select_os_volume(os_volume_size_gb, ironic_client, drac_client, node_uuid):
 
                     break
 
+            # Note: This code block represents single disk scenario.
+            raid_disk_ids = []
             if raid_size_gb == 0:
-                raise RuntimeError("There must be either a virtual disk that "
-                                   "is not a RAID 0 to install the OS on, or "
-                                   "os-volume-size-gb must be specified")
+                if virtual_disk_docs:
+                    for virtual_disk_doc in virtual_disk_docs:
+                        fqdd = get_fqdd(virtual_disk_doc, DCIM_VirtualDiskView)
+                        raid_type = utils.find_xml(
+                            virtual_disk_doc,
+                            'RAIDTypes',
+                            DCIM_VirtualDiskView).text
+
+                        if raid_type == RAID0:
+                            raid_size = get_size_in_bytes(virtual_disk_doc,
+                                                          DCIM_VirtualDiskView)
+                            raid_size_gb = int(raid_size) / units.Gi
+                            raid_disk_ids.append(raid_size_gb)
+                    if not len(raid_disk_ids) == 1:
+                        raise RuntimeError(
+                            "There must be either a virtual disk that "
```

Comment:
Can you change this error message to: "There must be a non-RAID0 virtual disk, a single disk RAID0, or a single JBOD disk to install the OS on, or os-volume-size-gb must be specified."

Reply:
Changed.